xml/hwbp_registers.xml

```diff
@@ -496,7 +496,11 @@
             trigger(s) matched.  If the bit is not implemented, it is always 0
             and writing it has no effect.
         </field>
-        <field name="0" bits="XLEN-7:10" access="R" reset="0" />
+        <field name="0" bits="XLEN-7:11" access="R" reset="0" />
+        <field name="nmi" bits="10" access="R/W" reset="0">
+            When this optional bit is set, non-maskable interrupts cause this
+            trigger to fire just as it does for other exceptions.
```

Comment:
I wonder if this bit belongs in the Interrupt Trigger register rather than Exception Trigger.  It seems like nmi is more like interrupts than it is like exceptions.

Is the intention to trigger on all nmi's regardless of M/S/U mode, or would the mode have to match an enabled mode to trigger?

I would suggest text along the lines of "When this optional bit is set, enable this trigger for non-maskable interrupts that are taken from M, S, or U mode."

Reply:
I assumed nmi would still depend on the m/s/u bits, for consistency with the other sources that can cause triggers in this register. It also affords a little more flexibility, although I'm not sure when somebody would want to trigger e.g. on NMIs from U mode but not other modes. I'm still leaning towards consistency. What do you think?

Reply:
Sounds like we should discuss at the next meeting.  NMIs "are only used for hardware error conditions" according to the Priv Spec, so it makes more sense to me to respond to any NMI regardless of mode enables.

Reply:
In today's meeting we agreed to ignore m/s/u for nmi. I've updated this PR.